src/Domain/Item/ItemService.ts

```diff
@@ -19,14 +21,23 @@ import { SaveItemsResult } from './SaveItemsResult'
 export class ItemService implements ItemServiceInterface {
   private readonly DEFAULT_ITEMS_LIMIT = 100000
   private readonly SYNC_TOKEN_VERSION = 2
-  private readonly MIN_CONFLICT_INTERVAL_MICROSECONDS = Time.MicrosecondsInAMillisecond
 
   constructor (
     @inject(TYPES.ItemRepository) private itemRepository: ItemRepositoryInterface,
     @inject(TYPES.Timer) private timer: TimerInterface
   ) {
   }
 
+  async computeIntegrityHash(userUuid: string): Promise<string> {
+    const timestampsInMicroseconds = await this.itemRepository.findDatesForComputingIntegrityHash(userUuid)
+
+    const timestampsInMilliseconds = timestampsInMicroseconds.map(timestampsInMicrosecond => Math.floor(timestampsInMicrosecond / Time.MicrosecondsInAMillisecond))
+
```

Comment:
how about `timestampMicro => Time.microToMilliseconds(timestampMicro)`?

Reply:
Will fix 👍

Reply:
@karolsojko  Wait here you are duplicating `timestampsInMicroseconds` as the array on line 32 and the mapped var on line 34. Not sure what kind of consequences this will have. Mapped var should be `timestampInMicroseconds`

Reply:
This is how this line looks now: `timestampsInMicroseconds.map(timestampsInMicroseconds => this.timer.convertMicrosecondsToMilliseconds(timestampsInMicroseconds))`

@standarius please add comments on the lines that are discussed - otherwise they don't get updated on the PR view.

@mobitar the timestamps taken to compute the integrity hash seem to be in milliseconds format `%Q`: https://github.com/standardnotes/syncing-server/blob/b6df312413b875c32b2ef7cb4fe4d9b6d78e4556/app/models/user.rb#L101

Reply:
@karolsojko ah, you mean if I don't comment after update, it still shows the file from before? huh, didn't know that. what about if I click resolve? that ok or blabla comments are needed for the robots?

Reply:
I think GH has a bug here where sometimes comments get detached from their thread.